dashboard/src/utils/filterDataset.js

```diff
@@ -1,8 +1,7 @@
-import { constructNewDate } from "./constructDate";
-
+import { constructUTCDate } from "./constructDate";
 export const filterData = (dataArray, startDate, endDate, controllerValue) => {
   let res = dataArray.filter((data) => {
-    let datasetDate = constructNewDate(data.metadata["dataset.created"]);
+    let datasetDate = constructUTCDate(new Date((data.metadata["dataset.created"]).split(":")[0]));
```

Comment:
There's no need to truncate the dataset timestamps to just the date.

Moreover, it is possible that, in the future, we'll want to be able to select things with a finer granularity than just a _day_.  (E.g., suppose I want only the runs which took place between 1:00 and 2:00?)  If we capture the full date-time here, we should be able to make that work with any selection range -- that is, we can worry about the granularity of the selectors elsewhere -- we don't have to worry about it here.

The server timestamps are currently UTC values, but they lack the timezone specifier.  For the moment (until we can get the server to produce full ISO 8601 format), I recommend that we just add a `Z` to the end of what we receive from the server and use the existing `Date()` constructor.  In the future, we can contemplate a more rigorous approach here.

Reply:
Can this be approach in a follow-up PR?

Reply:
Unless we are talking about deferring all of the time-based features, I would say "no".